Tests/OktaOAuth2Tests/JWTAuthorizationFlowTests.swift

```diff
@@ -0,0 +1,165 @@
+        urlSession.expect("https://example.okta.com/oauth2/v1/token",
+                          data: try data(from: .module, for: "token", in: "MockResponses"),
+                          contentType: "application/json")
+        
+        flow = client.jwtAuthorizationFlow()
+        
+        jwt = try JWT(JWT.mockIDToken)
+    }
+    
+    override func tearDownWithError() throws {
+        JWK.resetToDefault()
+        Token.resetToDefault()
+    }
+
+    func testWithDelegate() throws {
+        let delegate = JWTAuthorizationFlowDelegateRecorder()
+        flow.add(delegate: delegate)
+        
+        XCTAssertFalse(flow.isAuthenticating)
+        XCTAssertFalse(delegate.started)
+        
+        let expect = expectation(description: "Expected `start` succeeded")
+        flow.start(with: jwt) { result in
+            expect.fulfill()
+        }
+        
+        waitForExpectations(timeout: 5) { error in
+            XCTAssertNil(error)
+        }
+        
+        XCTAssertFalse(flow.isAuthenticating)
+        XCTAssertNotNil(delegate.token)
+        XCTAssertTrue(delegate.finished)
+        
+        XCTAssertEqual(urlSession.requests.count, 3)
+        
+        let request = try XCTUnwrap(urlSession.requests.first(where: { request in
+            request.url?.lastPathComponent == "token"
+        }))
+
+        XCTAssertEqual(request.url?.absoluteString, "https://example.okta.com/oauth2/v1/token")
+        XCTAssertEqual(request.bodyString, "assertion=\(JWT.mockIDToken)&client_id=clientId&grant_type=urn:ietf:params:oauth:grant-type:jwt-bearer&scope=profile+openid")
+    }
+    
+    func testAuthenticationSucceeded() throws {
+        let authorizeExpectation = expectation(description: "Expected `start` succeeded")
+        
+        XCTAssertFalse(flow.isAuthenticating)
+        
+        let expect = expectation(description: "start")
+        flow.start(with: jwt) { result in
+            switch result {
+            case .success:
+                XCTAssertFalse(self.flow.isAuthenticating)
+                
+                authorizeExpectation.fulfill()
+            case .failure(let error):
+                XCTFail(error.localizedDescription)
+            }
+            
+            expect.fulfill()
+        }
+        waitForExpectations(timeout: 1) { error in
+            XCTAssertNil(error)
+        }
+
+        XCTAssertFalse(flow.isAuthenticating)
+        
+        XCTAssertEqual(urlSession.requests.count, 3)
+        
+        let request = try XCTUnwrap(urlSession.requests.first(where: { request in
+            request.url?.lastPathComponent == "token"
+        }))
+
+        XCTAssertEqual(request.url?.absoluteString, "https://example.okta.com/oauth2/v1/token")
+        XCTAssertEqual(request.bodyString, "assertion=\(JWT.mockIDToken)&client_id=clientId&grant_type=urn:ietf:params:oauth:grant-type:jwt-bearer&scope=profile+openid")
+    }
+    
+#if swift(>=5.5.1) && !os(Linux)
```

Comment:
do we ci infra to test 'old swift' / iOS 12? Just curious

Reply:
No we don't. [Our support policy](okta-mobile-swift-task-cancellation) says that we support older versions, but it's on a best-effort basis.  e.g. we won't actively deprecate old versions if we don't have to, but if we find it necessary to do so (e.g. a new feature would be cumbersome to backport), then changing the minimum-supported OS version is considered a non-breaking change.

When this SDK was first written, even though Swift Concurrency was supported, there were bugs that prevented Xcode from compiling if the async operator were used. The Linux Swift runtime at the time also didn't support Swift Concurrency, so that's the reason for these `#if` checks. 

I really should pull them all out, and update the minimum Swift version (which is needed to add support for visionOS), but again, I'll wait for a major version change to do that.